main.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2019, Arm Limited and affiliates
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <string.h>
+
+#if defined(ATCA_HAL_I2C)
+#include "psa/crypto.h"
+#include "atecc608a_se.h"
+#include "atca_status.h"
+#include "atca_devtypes.h"
+#include "atca_iface.h"
+#include "atca_command.h"
+#include "atca_basic.h"
+#include "atca_helpers.h"
+
+#define ASSERT_STATUS(actual, expected)                             \
+    do                                                              \
+    {                                                               \
+        int ASSERT_STATUS_actual = (actual);                        \
+        int ASSERT_STATUS_expected = (expected);                    \
+        if ((ASSERT_STATUS_actual) != (ASSERT_STATUS_expected))     \
+        {                                                           \
+            printf("assertion failed at %s:%d "                     \
+                   "(actual=%d expected=%d)\n", __FILE__, __LINE__, \
+                    ASSERT_STATUS_actual, ASSERT_STATUS_expected);  \
+            return -1;                                              \
+        }                                                           \
+    } while(0)
+
```

Comment:
Suggestion: define and use `#define ASSERT_SUCCESS(expr) ASSERT_STATUS((expr), PSA_SUCCESS)`